### DISCUSSION TITLE
How does `neovis.js` display the names of nodes and edges?

### DISCUSSION THREAD

**Discussion by chenkuncloud** [neo4j-contrib/neovis.js#377] [Q&A] [answered]
Hi,

How does `neovis.js` display the names of nodes and edges? 

The name is displayed on the diagram of the demo, but the examples in the `/examples/` directory do not display the name.

My Neo4j Node's properties, I want to show `name`:
```text
<id>	0
name	abc
field2	value2
field3	value3
```

I only modified the serverUrl, serverUser, and serverPassword in [advanced-example.html](https://github.com/neo4j-contrib/neovis.js/blob/master/examples/advanced-example.html) locally, and the page effect is as follows:

![`%5)RF776`EDKLRGJGYKATO](https://github.com/neo4j-contrib/neovis.js/assets/40781280/c4c15ead-2093-4c81-bd4b-23008fc26320)

**Comment by exoup** [comment, marked as answer]
To see the `name` property on your node, you can try doing something like this:
```Javascript
                labels: {
                    User: {
                        label: "name"
                    }
                }
```

`name` can also be the name of any other property which a node possesses. In your example, `field2` or `field3`

To view edge labels, you need to do something like this:
```Javascript
                relationships: {
                    MY_RELATIONSHIP_LABEL: {
                        [NEOVIS_ADVANCED_CONFIG]: {
                            function: {
                                label: (edge) => edge.type
                            },
                        }
                    },
                },
```

You can also use the `label` property but it needs to be a property of the relationship, and there doesn't seem to be a way to call it with the `label` key, so you use the advanced config instead to return the edge type.

**Comment by chenkuncloud** [comment]
Thanks!

**Comment by xgh127** [comment]
Sorry，I still don't understand how to display the names of nodes.In this case:
`      labels: {
                    User: {
                        label: "name"
                    }
                }`
What does the "User" mean? What's the relationship between the so-called "User"  and my node ?Is this a properity of my node?

**Reply by thebestnom** [reply]
`User` here is the neo4j node label
